src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/IssueNumberRetriever.java

```diff
@@ -23,7 +27,18 @@ public IssueNumberRetriever(DataSource dataSource) {
     public long getNextIssueNumber() {
         try {
             Statement statement = dataSource.getConnection().createStatement();
-            ResultSet resultSet = statement.executeQuery(GET_NEXTVALUE_FROM_ISSUE_NUMBER_SEQUENCE);
+            ResultSet resultSet = statement.executeQuery(GET_NEXT_VALUE_FROM_ISSUE_NUMBER_SEQUENCE);
+            resultSet.next();
+            return resultSet.getLong("issue");
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    public long getCurrentIssueNumber() {
+        try {
+            Statement statement = dataSource.getConnection().createStatement();
+            ResultSet resultSet = statement.executeQuery(GET_CURRENT_VALUE_FROM_ISSUE_NUMBER_SEQUENCE);
             resultSet.next();
             return resultSet.getLong("issue");
```

Comment:
Same as above. `Statement` should always be closed after `ResultSet` it created is no longer needed. It will also automatically close its `ResultSet`.